sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -552,13 +548,24 @@ private EventHubConnectionProcessor buildConnectionProcessor(MessageSerializer m
         final String product = properties.getOrDefault(NAME_KEY, UNKNOWN);
         final String clientVersion = properties.getOrDefault(VERSION_KEY, UNKNOWN);
 
-        final Flux<EventHubAmqpConnection> connectionFlux = Mono.fromCallable(() -> {
-            final String connectionId = StringUtil.getRandomString("MF");
+        final Flux<EventHubAmqpConnection> connectionFlux = Flux.create(sink -> {
+            sink.onRequest(request -> {
+
+                if (request == 0) {
+                    return;
+                } else if (request > 1) {
+                    logger.warning("Requested more than one connection. Only emitting one. Request: {}", request);
+                }
```

Comment:
Can this happen? We always request 1 connection.

Reply:
No. should I throw an error instead if this ever happens? was trying to be defensive if we accidentally introduce a bug.